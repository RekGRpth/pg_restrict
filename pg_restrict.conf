shared_preload_libraries = 'pg_restrict'
pg_restrict.alter_system = on
pg_restrict.copy_program = on
pg_restrict.master_roles = 'role_restrict_1, postgres'
pg_restrict.nonremovable_databases = 'db_restrict_a, db_restrict_b, postgres'
pg_restrict.nonremovable_roles = 'role_restrict_a'
