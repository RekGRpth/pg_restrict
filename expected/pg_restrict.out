\set VERBOSITY terse
CREATE DATABASE db_restrict_1;
CREATE ROLE role_restrict_1 SUPERUSER LOGIN;	-- master role
CREATE ROLE role_restrict_2 SUPERUSER LOGIN;
CREATE ROLE role_restrict_3 SUPERUSER LOGIN;
CREATE TABLE tmp_restrict_1 (a integer);
INSERT INTO tmp_restrict_1 (a) VALUES(1),(2),(3);
\connect - role_restrict_2
CREATE DATABASE db_restrict_a;			-- non-removable database
DROP DATABASE db_restrict_a;			-- failed
ERROR:  cannot drop database "db_restrict_a"
DROP DATABASE db_restrict_1;			-- succeed
CREATE ROLE role_restrict_a;			-- non-removable role
DROP ROLE role_restrict_a;				-- failed
ERROR:  cannot drop role "role_restrict_a"
DROP ROLE role_restrict_3;				-- succeed
ALTER SYSTEM SET work_mem TO '2MB';		-- failed
ERROR:  cannot execute ALTER SYSTEM
COPY tmp_restrict_1 TO PROGRAM 'gzip > /tmp/tmp_restrict_1.dat.gz';		-- failed
ERROR:  cannot execute COPY ... PROGRAM
COPY tmp_restrict_1 TO '/tmp/tmp_restrict_2.dat';						-- succeed
\connect - role_restrict_1
DROP DATABASE db_restrict_a;			-- succeed
DROP ROLE role_restrict_a;				-- succeed
ALTER SYSTEM SET work_mem TO '3MB';		-- succeed
COPY tmp_restrict_1 TO PROGRAM 'gzip > /tmp/tmp_restrict_3.dat.gz';		-- succeed
\connect - euler
ALTER SYSTEM RESET work_mem;			-- succeed
DROP TABLE tmp_restrict_1;
DROP ROLE role_restrict_1;				-- succeed
DROP ROLE role_restrict_2;				-- succeed
